File: app/src/main/java/org/literacyapp/keyboard/receiver/StudentUpdatedReceiver.java
package org.literacyapp.keyboard.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.util.Log;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Set;

public class StudentUpdatedReceiver extends BroadcastReceiver {

    public static final String PREF_STUDENT_LETTERS = "pref_student_letters";
    public static final String PREF_STUDENT_NUMBERS = "pref_student_numbers";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(getClass().getName(), "onReceive");

        // Customize the user interface to match the current Student's level

        ArrayList<String> availableLetters = intent.getStringArrayListExtra("availableLetters");
        Log.i(getClass().getName(), "availableLetters: " + availableLetters);

        ArrayList<String> availableNumbers = intent.getStringArrayListExtra("availableNumbers");
        Log.i(getClass().getName(), "availableNumbers: " + availableNumbers);

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);

        if (availableLetters != null) {
            Set<String> availableLetterSet = new HashSet<>();
            for (String availableLetter : availableLetters) {
                availableLetterSet.add(availableLetter);
            }
            Log.i(getClass().getName(), "Storing availableLettersSet: " + availableLetterSet);
            sharedPreferences.edit().putStringSet(PREF_STUDENT_LETTERS, availableLetterSet).commit();
        }

        if (availableNumbers != null) {
            Set<String> availableNumberSet = new HashSet<>();
            for (String availableNumber : availableNumbers) {
                availableNumberSet.add(availableNumber);
            }
            Log.i(getClass().getName(), "Storing availableNumbersSet: " + availableNumberSet);
            sharedPreferences.edit().putStringSet(PREF_STUDENT_NUMBERS, availableNumberSet).commit();
        }
    }
}

File: app/src/main/java/ai/elimu/keyboard/receiver/StudentUpdatedReceiver.kt
package ai.elimu.keyboard.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.preference.PreferenceManager
import android.util.Log
import androidx.core.content.edit

class StudentUpdatedReceiver : BroadcastReceiver() {
    
    private val TAG = javaClass.getName()
    
    override fun onReceive(context: Context?, intent: Intent) {
        Log.i(TAG, "onReceive")

        // Customize the user interface to match the current Student's level
        val availableLetters = intent.getStringArrayListExtra("availableLetters")
        Log.i(TAG, "availableLetters: $availableLetters")

        val availableNumbers = intent.getStringArrayListExtra("availableNumbers")
        Log.i(TAG, "availableNumbers: $availableNumbers")

        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

        if (availableLetters != null) {
            val availableLetterSet: MutableSet<String?> = HashSet<String?>()
            for (availableLetter in availableLetters) {
                availableLetterSet.add(availableLetter)
            }
            Log.i(TAG, "Storing availableLettersSet: $availableLetterSet")
            sharedPreferences.edit(commit = true) {
                putStringSet(
                    PREF_STUDENT_LETTERS,
                    availableLetterSet
                )
            }
        }

        if (availableNumbers != null) {
            val availableNumberSet: MutableSet<String?> = HashSet<String?>()
            for (availableNumber in availableNumbers) {
                availableNumberSet.add(availableNumber)
            }
            Log.i(TAG, "Storing availableNumbersSet: $availableNumberSet")
            sharedPreferences.edit(commit = true) {
                putStringSet(
                    PREF_STUDENT_NUMBERS,
                    availableNumberSet
                )
            }
        }
    }

    companion object {
        const val PREF_STUDENT_LETTERS: String = "pref_student_letters"
        const val PREF_STUDENT_NUMBERS: String = "pref_student_numbers"
    }
}
